lang/c/src/datafile.c

```diff
@@ -41,6 +43,15 @@ struct avro_file_reader_t_ {
 	char * current_blockdata;
 };
 
+struct avro_file_buffer_reader_t_ {
```

Comment:
This file reader structure was added to avoid usage of block memory reader. It loads the whole block of the file to ram. As our writer schema (it contains in the avro file) is nested. All data encapsulated in the single record -> we have one huge allocation in size of the file.